registry/cache.go

```diff
@@ -14,28 +14,32 @@ import (
 )
 
 type Cache struct {
-	backend
+	next   dockerRegistryInterface
 	creds  Credentials
 	expiry time.Duration
 	Client MemcacheClient
 	logger log.Logger
 }
 
-func NewCache(b backend, creds Credentials, cache MemcacheClient, expiry time.Duration, logger log.Logger) *Cache {
-	return &Cache{
-		backend: b,
-		creds:   creds,
-		expiry:  expiry,
-		Client:  cache,
-		logger:  logger,
+type CachedDockerRegistry func(dockerRegistryInterface) dockerRegistryInterface
+
+func NewCache(creds Credentials, cache MemcacheClient, expiry time.Duration, logger log.Logger) CachedDockerRegistry {
```

Comment:
We always call `NewCache(...)(client)`, so let's take `client` as a regular argument, and make it:
```
func NewCache(next dockerRegistryInterface, creds Credentials, expiry time.Duration, Client MemcacheClient, logger log.Logger) dockerRegistryInterface
```